catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/monitor/impl/SubscriptionsPersistentStoreImpl.java

```diff
@@ -0,0 +1,278 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.monitor.impl;
+
+import static org.apache.commons.lang3.Validate.notBlank;
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.codice.ddf.catalog.ui.query.monitor.api.SubscriptionsPersistentStore;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Associate sets of emails with an ID. This implementation will preserve any other
+ * properties stored under the ID.
+ */
+public class SubscriptionsPersistentStoreImpl implements SubscriptionsPersistentStore {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(SubscriptionsPersistentStore.class);
+
+    private static final String EMAIL_PROPERTY = "emails";
+
+    private static final String ID = "id";
+
+    private static final Lock LOCK = new ReentrantLock();
+
+    private final PersistentStore persistentStore;
+
+    /**
+     * @param persistentStore must be non-null
+     */
+    public SubscriptionsPersistentStoreImpl(PersistentStore persistentStore) {
+        notNull(persistentStore, "persistentStore must be non-null");
+        this.persistentStore = persistentStore;
+    }
+
+    private List<Map<String, Object>> get(String id) throws PersistenceException {
+        LOCK.lock();
+        try {
+            List<Map<String, Object>> results = persistentStore.get(SubscriptionsPersistentStore.SUBSCRIPTIONS_TYPE, query(id));
+            assert results.size() <= 1;
+            return results;
+        } finally {
+            LOCK.unlock();
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void addEmails(String id, Set<String> emails) {
+        notBlank(id, "id must be non-blank");
+        notNull(emails, "emails must be non-null");
+        emails.forEach(email -> notBlank(email, "emails in set must be non-blank"));
+
+        LOCK.lock();
+        try {
+
+            try {
```

Comment:
I don't think this needs to be a nested `try-catch`. The `catch` could be moved before the `finally`.